aim/web/ui/src/utils/app/onAxesScaleRangeChange.ts

```diff
@@ -0,0 +1,52 @@
+import _ from 'lodash-es';
+
+import { ANALYTICS_EVENT_KEYS } from 'config/analytics/analyticsKeysMap';
+
+import * as analytics from 'services/analytics';
+
+import { IModel, State } from 'types/services/models/model';
+import { IAxesScaleRange } from 'types/components/AxesPropsPopover/AxesPropsPopover';
+
+import updateURL from './updateURL';
+
+const onAxesScaleRangeChange = _.debounce(
+  <M extends State>({
+    range,
+    model,
+    appName,
+  }: {
+    range: Partial<IAxesScaleRange>;
+    model: IModel<M>;
+    appName: string;
+  }): void => {
+    let configData = model?.getState()?.config;
+    if (configData?.chart) {
+      configData = {
+        ...configData,
+        chart: {
+          ...configData.chart,
+          axesScaleRange: {
+            ...configData.chart.axesScaleRange,
+            yAxis: {
+              ...configData.chart.axesScaleRange.yAxis,
+              ...(range.yAxis || {}),
+            },
+            xAxis: {
+              ...configData.chart.axesScaleRange.xAxis,
+              ...(range.xAxis || {}),
+            },
+          },
+        },
+      };
+      model.setState({ config: configData });
+      updateURL({ configData, appName });
+    }
+    analytics.trackEvent(
+      // @ts-ignore
```

Comment:
Why are we using ts-ignore here?

Reply:
since we didn't add types for the analytics config